internal/controller/sync/controller.go

```diff
@@ -152,8 +158,22 @@ func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 		return reconcile.Result{}, nil
 	}
 
+	syncContext := sync.NewContext(ctx, wsCtx)
+
+	// if desired, fetch the cluster path as well (some downstream service providers might make use of it,
+	// but since it requires an additional permission claim, it's optional)
+	if r.pubRes.Spec.EnableClusterPaths {
+		lc := &kcpdevcorev1alpha1.LogicalCluster{}
+		if err := r.vwClient.Get(wsCtx, types.NamespacedName{Name: kcpdevcorev1alpha1.LogicalClusterName}, lc); err != nil {
+			return reconcile.Result{}, fmt.Errorf("failed to retrieve remote logicalcluster: %w", err)
+		}
+
+		path := lc.Annotations[kcpcore.LogicalClusterPathAnnotationKey]
+		syncContext = syncContext.WithClusterPath(logicalcluster.NewPath(path))
```

Comment:
Should we handle the case that this annotation is empty and throw an error? Is this a valid or invalid scenario?